### PR TITLE
UI for Data Publication by year

### LINKED ISSUE #638: Update Publication UI for Final (2018) and Ultimate (2017) datasets
Short description explaining the high-level reason for the new issue.

## Current behavior
[Data-publication](https://ffiec.cfpb.gov/data-publication/) does not currently have a section for `Final` or `Ultimate` datasets.

## Expected behavior
The Data Publication page will have UI navigation available that include `Final` and `Ultimate` HMDA datasets: LAR, TS, Panel.

**Options**
1) Create a new page for both `Final` and `Ultimate` similar to `Dynamic` or `[Snapshot](https://ffiec.cfpb.gov/data-publication/snapshot-national-loan-level-dataset)`. This new page would be linked on the [Data-publication](https://ffiec.cfpb.gov/data-publication/) page similar to the `Snapshot` or `Dynamic` pages.

2) Modify the UI so that there is a selector for `Snapshot`, `Dynamic`, `Final`, `Ultimate` similar to the year selector currently in place on the `[Snapshot](https://ffiec.cfpb.gov/data-publication/snapshot-national-loan-level-dataset)` and `Dynamic` pages. This selector would load a page that is specific to the dataset version and allow a user to select the year of interest.

- [ ] Draft text to explain the differences between each dataset. This text would be present on the appropriate page similar to the text on the `[Snapshot](https://ffiec.cfpb.gov/data-publication/snapshot-national-loan-level-dataset)` page.

### LINKED ISSUE #1315: Include the size of the downloaded dataset prior to downloading data publication products
It would be helpful if the size of the dataset is shown prior to downloading so I can estimate exactly how large the file is before downloading. It would be also be helpful to have the download available as a txt or flat file to reduce the file time for downloading.

### PR DESCRIPTION
**Pull request by meissadia**
Closes #638 
Closes #1315

# Changes
- Adds year selector to enable display of different products for each year
- Creates a configuration where each year's available publications can be defined
- Adds the One Year and Three Year dataset views

# Todo
- [x] One Year add links
- [x] Three Year (2017, 2018) add links
- [x] Update S3 bucket names from one/three year
- Tests - Moved to https://github.com/cfpb/hmda-frontend/issues/1450

# Screenshots

## All Publications
all-pubs

## One Year Dataset

|year|screenshot|
|---|---|
|2017|one-year-2017|
|2018|one-year-2018|

## Three Year Dataset
three-year

### CODE REVIEW

**Comment by meissadia** [comment]
`common/S3Integrations`, `common/LargeFileWarning`
- http://localhost:3000/data-publication/snapshot-national-loan-level-dataset/2020

`common/constants/`
  - New configuration to show different years of availability
  - Shows more years - http://localhost:3000/data-publication/snapshot-national-loan-level-dataset/2020
  - Shows only 2019 & 2020 - http://localhost:3000/data-publication/snapshot-national-loan-level-dataset/2020

`common/s3/`
  - Explores RTK to manage state around S3 header information

`src/data-browser/datasets`
  - Integrates the LargeFileWarning component (!Warning: Large File)
  - http://localhost:3000/data-publication/snapshot-national-loan-level-dataset/2021 

`src/data-publication/Home.jsx`
  - Adds a year selector to display the different datasets by year, based on the `constants/publications-by-year.js` configuration.
  - http://localhost:3000/data-publication/2021

`src/data-publication/index.js`
  - Routing for the new One Year / Three Year pages

`src/data-publication/constants/one-year-datasets.js`
`src/data-publication/constants/three-year-datasets.js`
`src/data-publication/constants/snapshot-datasets.js`
  - The S3 files that each dataset's page will link to
  - i.e. The left column on http://localhost:3000/data-publication/three-year-national-loan-level-dataset/2018

`src/data-publication/reports/one_year`
`src/data-publication/reports/three_year`
  - Pages providing the details for the new datasets, links to the dataset files, and links to documentation
  - http://localhost:3000/data-publication/one-year-national-loan-level-dataset/2020
  - http://localhost:3000/data-publication/three-year-national-loan-level-dataset/2018

`src/homepage/index.js`
  - Adds direct links under the `Data Publications` section for the new datasets
  - http://localhost:3000/

File: src/data-publication/constants/snapshot-dataset.js [hunk 1]
```diff
@@ -6,9 +6,34 @@ const ARID2017_XREF = {
 }
 
 export const SNAPSHOT_DATASET = {
+  2021: {
+    snapshot_date: 'May 4, 2022',
+    datasets: [
+      {
+        csv: 'https://s3.amazonaws.com/cfpb-hmda-public/prod/snapshot-data/2020/2020_public_lar_csv.zip',
```
Comment (meissadia) [inline src/data-publication/constants/snapshot-dataset.js:13]: Update to 2021

**Comment by Michaeldremy** [comment]
Changes look good to me!

File: src/data-publication/constants/snapshot-dataset.js (hunk 1 above)
Comment (meissadia) [reply src/data-publication/constants/snapshot-dataset.js:13]: Addressed in follow-up PR for adding 2021 data: https://github.com/cfpb/hmda-frontend/pull/1436/files#diff-3a33bfdc7449537e15122a081cfb9b8249a48e51f3bc5f96019895c644a8c39dR14

**Pull request merged**